vite.config.ts

```diff
@@ -4,25 +4,48 @@ import { VitePWA } from 'vite-plugin-pwa'
 import react from '@vitejs/plugin-react'
 import pwaManifestConfig from './manifest.config.js'
 import tsconfigPaths from 'vite-tsconfig-paths'
+import { loadEnv } from 'vite'
 import { defineConfig } from 'vitest/dist/config.js'
+import viteRollbar from 'vite-plugin-rollbar'
 
 // References:
 // https://vitejs.dev/config/
 // https://vite-pwa-org.netlify.app/assets-generator/#example-using-minimal-preset
-export default defineConfig({
-  plugins: [tsconfigPaths(), react(), VitePWA(pwaManifestConfig)],
-  test: {
-    globals: true,
-    environment: 'jsdom',
-    setupFiles: ['src/test/setup.ts'],
-  },
-  server: {
-    proxy: {
-      '/api': {
-        target: 'http://localhost:3000',
-        changeOrigin: false,
-        rewrite: (path) => path.replace(/^\/api/, ''),
+export default defineConfig(({ mode }) => {
+  const env = loadEnv(mode, process.cwd(), '')
```

Comment:
This is the `defineConfig` syntax when environment variables need to be loaded. Since it is a function, it also allows for dynamically constructing the config, as seen here. It won't add the viteRollbar plugin unless ROLLBAR_ACCESS_TOKEN is defined in the environment.

https://vitejs.dev/config/#using-environment-variables-in-config